Document libvirt dnsmasq for DNS of non-.local domains

paususe opened the pull request:
## What does this PR change?

Document how to use libvirt's dnsmasq for DNS resolution.

This is a must when there are Windows guests, as mDNS is broken on Windows due to wrong decisions by Microsoft back in the day.

rjmateus commented on backend_modules/libvirt/README.md:
````diff
@@ -148,3 +148,193 @@ module "base" {
+But only for the `suse.lab` domain:
+
+```
+$ sudo vi /etc/NetworkManager/dnsmasq.d/libvirt_dnsmasq.conf 
````
```suggestion
$ sudo vi /etc/NetworkManager/dnsmasq.d/libvirt_dnsmasq.conf
```
remove white space at the end of the line

paususe replied:
Done

paususe commented on backend_modules/libvirt/README.md:
```diff
@@ -148,3 +148,193 @@ module "base" {
+
+### Configuring NetworkManager
+
+If you are using NetworkManager, tell it to control dnsmasq:
```
It's not on the guest machine but on the host machine.

paususe commented on backend_modules/libvirt/README.md:
````diff
@@ -148,3 +148,193 @@ module "base" {
+It will look like this:
+
+```xml
+<network connections='3'>
````
Done

paususe commented on backend_modules/libvirt/README.md:
````diff
@@ -148,3 +148,193 @@ module "base" {
+You can now add the MACs and desired IPs in the `ip` block, below the `range` entry. Please note I have changed the range start address to allocate space for static address leases. Your XML will look like this:
+
+```xml
+<ip address='192.168.122.1' netmask='255.255.255.0'>
````
Done

paususe commented on backend_modules/libvirt/README.md:
````diff
@@ -148,3 +148,193 @@ module "base" {
+We could add the hostnames to the XML too but in that case, name resolution would only work across virtual guest. As we want name resolution to work also between host and guest, we will now add the domain name to the XML, right after the `mac address` block. In the end, your XML will look like this:
+
+```xml
+<network connections='3'>
````
Done

paususe commented on backend_modules/libvirt/README.md:
````diff
@@ -148,3 +148,193 @@ module "base" {
+</network>
+```
+
+Now edit `/etc/hosts` and add your guests:
````
This is on the host machine, not on the guest machine. What you are doing with this step and previous one is associating an IP and DNS A register to a MAC and that happens on the host.

The reason I am listing this here, and not at the end, is because this way it's self-contained: everything in Configuring Network Manager (or Alternative), and the "Finalize and test everything works as expected") needs to be done only the first time.

You could argue the `virsh net-destroy` could be placed before editing `/etc/hosts` and I may even agree (although I'd rather have all the changes on the host done, then apply them with net-destroy and net-start, than finish the MAC+IP association and leave the door open to forgetting hostname+IP association).

paususe commented on backend_modules/libvirt/README.md:
````diff
@@ -148,3 +148,193 @@ module "base" {
+And now you will need to destroy and start again your network:
+
+```
+$ sudo virsh net-destroy default && virsh net-start default
````
Done

paususe commented on backend_modules/libvirt/README.md:
````diff
@@ -148,3 +148,193 @@ module "base" {
+If you are using the `default`virtual network in the 192.168.122.1 network, you will have this interface:
+
+```
+$ virsh net-list --all
````
Done

paususe commented on backend_modules/libvirt/README.md:
````diff
@@ -148,3 +148,193 @@ module "base" {
+</ip>
+```
+
+We could add the hostnames to the XML too but in that case, name resolution would only work across virtual guest. As we want name resolution to work also between host and guest, we will now add the domain name to the XML, right after the `mac address` block. In the end, your XML will look like this:
````
Done

paususe commented:
@rjmateus 
I have fixed all issues and also added an explicit mention to how to disable Avahi and set the domain in main.tf, which was obviated because this was already explained in README.md.

rjmateus commented on backend_modules/libvirt/README.md:
````diff
@@ -148,3 +148,221 @@ module "base" {
+server=/suse.lab/192.168.122.1
+```
+
+### Alternative: no NetworkManager
````
I have tested the Alternative: no NetworkManager and looks good to me. 
I have a newby question (that could make sense document it) : How can I cheeck if I'm using network Manager to configure dnsmasq?
To configure my machine I follow the tutorial in: https://wiki.suse.net/index.php?title=Services_Team/Policies/openVPN/client_setup

paususe replied:
@rjmateus 
I have added how to check for NetworkManager.

NetworkManager, OpenVPN and dnsmasq are three completely orthogonal tools. They can be combined (typical these days, the main use case), or you may run e. g. wicked + OpenVPN from the command-line + external dnsmasq (manually configured and managed). 

I don't think there's a need to document anymore here "just in case", let's see if  someone comes with an actual need for an alternative configuration.
